lib/public-api.js

```diff
@@ -1,5 +1,21 @@
 "use strict";
 
-module.exports = require("./DOMException").interface;
+const DOMException = require("./DOMException");
 
-Object.setPrototypeOf(module.exports.prototype, Error.prototype);
+// Special install function to make the DOMException inherit from Error.
+// https://heycam.github.io/webidl/#es-DOMException-specialness
+function installDOMException(globalObject) {
+  DOMException.install(globalObject);
+
+  if (!globalObject.Error || typeof globalObject.Error !== "function") {
```

Comment:
When attempting to integrate this change in jsdom, I found out that the `window` object doesn't expose the primordials and the following branch check fails. I also realized that some built-ins (like `ArrayBuffer`) are attached but there is a [TODO](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/browser/Window.js#L439-L449) to remove them which is strange. 

IMO, we should remove the comment and expose all the built-ins. Thoughts?

Reply:
Hmm hmm. So what's going on here is that if you don't enable scripting in jsdom, then none of the ECMAScript globals get installed. I guess we alias some of them to the outer globals anyway to make some tests pass. If you do enable scripting, then they should all be installed by the vm module.

I think what we should do is either:

- Remove support for no-scripting from jsdom. This ensures that consistently all jsdom windows have non-shared globals, no matter whether those are web globals or ECMAScript globals.

- Change our strategy to branch on scripting support. If scripting is not enabled, then alias all the ECMAScript globals (not just the small list we do today). If it is enabled, then alias none of them.

The drawback of the first option here is the potential performance hit for the cases where people don't need to run scripting. The drawback of the second is inconsistency.